src/web_interface/components/analysis_routes.py

```diff
@@ -165,5 +168,8 @@ def _schedule_re_analysis_task(self, uid, analysis_task, re_do):
             sc.add_re_analyze_task(fw)
 
     @roles_accepted(*PRIVILEGES['delete'])
+    @AppRoute('/admin/re-do_analysis/<uid>', GET, POST)
     def _re_do_analysis(self, uid):
-        return self._update_analysis(uid, re_do=True)
+        if request.method == POST:
```

Comment:
I though such distinction could now be omitted by directly labeling the functions with correct HTTP method?

Reply:
Yes we could do that, but in this special case I found it to be more compact this way, because this is more or less only a redirect to the update analysis view functions. I could change it if you want, though.

Reply:
I get what you are saying. I'm looking at it right now and don't really feel one is much better than the other. So this can stay as it is.